schema["type"] should allow str or unicode, and the error now prints the type

MasterEric opened the pull request:
A fix for a frustrating bug I encountered. I'm surprised the unit tests even work.

Grokzen commented:
@MasterEric You broke the tests, needs to be fixed and the linting of the changed code is not good. Both needs to be fixed before i could accept this PR.

Grokzen commented on pykwalify/rule.py:
```diff
@@ -380,9 +380,9 @@ def init(self, schema, path):
                     self.type = t
             else:
-                if not isinstance(schema["type"], str):
+                if not (isinstance(schema["type"], str) or isinstance(schema["type"], unicode)):
```
You glob them together into into something like
```
if not isinstance(schema["type"], (str, int))
```
and get the same logic and much cleaner code.

Grokzen commented on pykwalify/rule.py:
```diff
@@ -380,9 +380,9 @@ def init(self, schema, path):
+                if not (isinstance(schema["type"], str) or isinstance(schema["type"], unicode)):
                     raise RuleError(
-                        msg=u"Key 'type' in schema rule is not a string type",
+                        msg=u"Key 'type' in schema rule is not a string type (found "+str(type(schema["type"]))+")",
```
Horrible python linting, should be
```
msg=u"Key 'type' in schema rule is not a string type (found " + str(type(schema["type"])) + ")",
```

MasterEric commented:
Apologies for the delay in fixing your issues.

I corrected the linting on my code changes, and discovered that modifying one of the error messages to be more verbose broke an assertion in one of the unit tests, which I fixed. All unit tests now pass on my machine.

MasterEric commented:
@Grokzen Looking at the Travis page, it says that 2 checks failed, however there are no logs for the checks that failed. All tests where there are logs show all checks have passed. Can you please fix this issue?

MasterEric commented:
@Grokzen This issue appears to have been closed without comment, can you please clarify?

Grokzen commented:
@MasterEric I accidentally closed all open MR:s when i removed the destination branch. I am currently manually merging all of them including this one.

Grokzen commented:
@MasterEric Manually merged the commits here 24adf3d1a1d74ccc260b0eca1a87e1ea48a3785b..dfb45cb6450e00f80dcfe0f05dad2e1b7cd9e9f0
